src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,240 @@
+
+from code42cli.click_ext.groups import OrderedGroup
+from code42cli.options import format_option
+from code42cli.options import sdk_options
+from code42cli.options import set_begin_default_dict
+from code42cli.options import set_end_default_dict
+from code42cli.output_formats import OutputFormatter
+
+
+case_number_arg = click.argument("case-number", type=int)
+name_option = click.option("--name", help="Name of the case.",)
+assignee_option = click.option("--assignee", help="User UID of the assignee.")
+description_option = click.option("--description", help="Description of the case.")
+notes_option = click.option("--notes", help="Notes on the case.")
+subject_option = click.option("--subject", help="User UID of a subject of the case.")
+status_option = click.option(
+    "--status",
+    help="Status of the case. `OPEN` or `CLOSED`.",
+    type=click.Choice(CaseStatus.choices()),
+)
+file_event_id_option = click.option(
+    "--event-id", required=True, help="File event id associated to the case."
+)
+
+CASES_KEYWORD = "cases"
+BEGIN_DATE_DICT = set_begin_default_dict(CASES_KEYWORD)
+END_DATE_DICT = set_end_default_dict(CASES_KEYWORD)
+
+
+def _get_cases_header():
+    return {
+        "number": "number",
+        "name": "Name",
+        "assignee": "Assignee",
+        "status": "Status",
+        "createdAt": "Creation Time",
+        "findings": "Notes",
+    }
+
+
+def _get_events_header():
+    return {
+        "eventId": "Event Id",
+        "eventTimestatmp": "Timestamp",
+        "filePath": "Path",
+        "fileName": "File",
+        "exposure": "Exposure",
+    }
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def cases(state):
+    """For managing cases and events associated with cases."""
+    pass
+
+
+@cases.command()
+@click.argument("name")
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@sdk_options()
+def create(state, name, subject, assignee, description, notes):
+    """Create a new case."""
+    state.sdk.cases.create(
+        name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+    )
+
+
+@cases.command()
+@case_number_arg
+@name_option
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@status_option
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+@cases.command("list")
+@click.option(
+    "--name", help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--subject", help="Filter by user UID of the subject of a case.")
+@click.option("--assignee", help="Filter by user UID of assignee.")
+@click.option("--begin-create-time", **BEGIN_DATE_DICT)
```

Comment:
Do we need to support --checkpoint feature here?

Reply:
I refactored (not exactly) the "begin option" to be used without extractor!

Reply:
I don't think we need a checkpoint here.